From e9f0cde11f06ea2c64d8d6d5db61cdf387af2980 Mon Sep 17 00:00:00 2001
From: Daniel Guettler <daniel.guettler@gmail.com>
Date: Mon, 26 Jan 2009 11:28:34 -0500
Subject: [PATCH] check for template with specified extension but without template handler extension
 e.g. render :file => 'test/hello_world.js' should render the hello_world.js template
 if it exists instead raising a missing template exception

---
 actionpack/lib/action_view/base.rb           |    2 ++
 actionpack/test/fixtures/test/hello_world.js |    1 +
 actionpack/test/template/render_test.rb      |    4 ++++
 3 files changed, 7 insertions(+), 0 deletions(-)
 create mode 100644 actionpack/test/fixtures/test/hello_world.js

diff --git a/actionpack/lib/action_view/base.rb b/actionpack/lib/action_view/base.rb
index f5f3fda..dbbc159 100644
--- a/actionpack/lib/action_view/base.rb
+++ b/actionpack/lib/action_view/base.rb
@@ -320,6 +320,8 @@ module ActionView #:nodoc:
         # OPTIMIZE: Checks to lookup template in view path
         if template = self.view_paths["#{template_file_name}.#{template_format}"]
           template
+        elsif template_file_extension && template = self.view_paths["#{template_file_name}.#{template_file_extension}"]
+          template
         elsif template = self.view_paths[template_file_name]
           template
         elsif (first_render = @_render_stack.first) && first_render.respond_to?(:format_and_extension) &&
diff --git a/actionpack/test/fixtures/test/hello_world.js b/actionpack/test/fixtures/test/hello_world.js
new file mode 100644
index 0000000..2f3f65f
--- /dev/null
+++ b/actionpack/test/fixtures/test/hello_world.js
@@ -0,0 +1 @@
+var greeting = 'Hallo World!';
\ No newline at end of file
diff --git a/actionpack/test/template/render_test.rb b/actionpack/test/template/render_test.rb
index 531df6c..3872ebf 100644
--- a/actionpack/test/template/render_test.rb
+++ b/actionpack/test/template/render_test.rb
@@ -34,6 +34,10 @@ class ViewRenderTest < Test::Unit::TestCase
     assert_equal "Hello world!", @view.render(:file => template_path)
   end
 
+  def test_render_file_not_using_template_handler_extension
+    assert_equal "var greeting = 'Hallo World!';", @view.render(:file => 'test/hello_world.js')
+  end
+
   def test_render_file_with_instance_variables
     assert_deprecated do
       assert_equal "The secret is in the sauce\n", @view.render("test/render_file_with_ivar.erb")
-- 
1.5.6

